tools/validate_configs/test_configs/spack-environments.yaml

```diff
@@ -60,22 +60,36 @@ deployment_groups:
               - - $mpi_packages
               - - $%compilers
               - - $^mpis
-      - name: 'pkg-test'
-        packages:
-        - intel-mpi@2018.4.274 %gcc@10.3.0
-        - gromacs@2021.2 %gcc@10.3.0 ^intel-mpi@2018.4.274
-      - name: 'mixed-test'
+      - destination: /apps/spack/mixed_env_file.yaml
         content: |
           spack:
             specs:
               intel-mpi@20184.274 %gcc@10.3.0
-        packages:
-        - gromacs@2021.2 %gcc@10.3.0 ^intel-mpi@2018.4.274
-      - name: 'empty-test'
       gpg_keys:
       - type: 'file'
         path: '/tmp/spack_key.gpg'
       commands: |
+        if ! spack env list | grep -q test-file; then
```

Comment:
This is likely intentional, but I wanted to point out that with this pattern you're also ignoring changes in the data file if the environment has been set up once.

Reply:
Yes. This is parity (or better) than what we were doing before except that now it is more visible to user.

I did not see any way to tell spack "make this already created file look like this yaml definition". Another option would be to always remove and create the env but I suspect that would be lots of needless work in most cases.

My personal recommendation would be to use this if to create a named env without a config. Then use spack add to build the environment as spack add is tolerant if the spec is already added. This way it would be idempotent.